server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -163,7 +163,7 @@ private static void handleFieldWildcards(IndexShard indexShard, TermVectorsReque
     private static boolean isValidField(MappedFieldType fieldType) {
         // must be a string
         if (fieldType instanceof KeywordFieldMapper.KeywordFieldType == false
-                && fieldType instanceof TextFieldMapper.TextFieldType == false) {
+                && fieldType instanceof StringFieldType == false) {
```

Comment:
looks like we can remove KeywordFieldType from the list since it extends StringFieldType?

Reply:
Doh!